linkerd/stack/src/fallback.rs

```diff
@@ -172,3 +199,57 @@ where
         }
     }
 }
+
+// === impl ReadyFallback ===
+
+impl<N, T, A, B, P, Req> tower::Service<Req> for ReadyFallback<N, T, A, B, P>
+where
+    A: tower::Service<Req>,
+    A::Error: Into<Error>,
+    N: NewService<T, Service = B>,
+    B: tower::Service<Req, Response = A::Response>,
+    B::Error: Into<Error>,
+    P: Fn(&Error) -> bool,
+{
+    type Response = A::Response;
+    type Error = Error;
+    type Future = Either<A::Future, B::Future>;
+
+    fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        // Fallback may only be set until the initial state is polled to ready.
+        match self.fallback {
+            None => self.inner.poll_ready(cx),
+            ref mut fallback => {
+                debug_assert!(
+                    matches!(self.inner, Either::A(_)),
+                    "must not be in fallback yet"
+                );
+
+                match futures::ready!(self.inner.poll_ready(cx)) {
+                    Ok(()) => {
+                        // If the primary service becomes ready, drop the
+                        // fallback.
+                        drop(fallback.take().expect("fallback expected"));
+                        Poll::Ready(Ok(()))
+                    }
```

Comment:
so, we won't use the fallback on errors that occur after the primary becomes ready a single time? i guess that makes sense in the case where the fallback is used for stuff like a discovery lookup returning `InvalidArgument`, or when a port is skipped, or whatever, but it seems surprising in the general case. this is probably right for what we're currently using `Fallback` for, but it might be worth writing this behavior down so we don't use fallback in cases where that doesn't make sense in the future?